### PR TITLE
Move SpeciesSet from Utilities to Particle

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
triggered by the move, add more cleaning to SpinDensityNew.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-sever

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted

### CODE REVIEW

File: src/Estimators/tests/test_SpinDensityNew.cpp [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #include "ValidSpinDensityInput.h"
 #include "SpinDensityNew.h"
 #include "RandomForTest.h"
-#include "Utilities/SpeciesSet.h"
+//#include "Utilities/SpeciesSet.h"
```
Comment (PDoakORNL) [inline src/Estimators/tests/test_SpinDensityNew.cpp:19]: should be able to just delete now.

**Review by PDoakORNL — APPROVED** [review]
I think this makes more sense. And thanks for all the formatting.

File: src/Estimators/tests/test_SpinDensityNew.cpp (hunk 1 above)
Comment (ye-luo) [reply src/Estimators/tests/test_SpinDensityNew.cpp:19]: Will handle it when I deal with merge conflicts.

**Comment by ye-luo** [comment]
Start testing in-house

**Pull request merged**